lib/spack/spack/cmd/buildcache.py

```diff
@@ -59,6 +59,9 @@ def setup_parser(subparser):
     create.add_argument('-d', '--directory', metavar='directory',
                         type=str, default='.',
                         help="directory in which to save the tarballs.")
+    create.add_argument('-s', '--skip_rebuild_index', action='store_true',
```

Comment:
Can this be: `--no-rebuild-index` and have no short argument?  Seems like it would be used infrequently or only in scripts (where you want readability), but maybe I'm underestimating.  Also I dunno why `allow_root` has an underscore -- but nearly all the other commands in Spack use `-`.